proxy/http/ForwardedConfig.cc

```diff
@@ -0,0 +1,126 @@
+/** @file
+
+  Configuration of Forwarded HTTP header option.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <bitset>
+#include <string>
+#include <cctype>
+
+#include <ts/string_view.h>
+#include <ts/MemView.h>
+
+#include <HttpConfig.h>
+
+namespace HttpForwarded
+{
+OptionBitSet
+optStrToBitset(ts::string_view optConfigStr, std::string &error)
```

Comment:
Now that I see this, I think this should be

```
optStrToBitset(ts::string_view optConfigStr, const char **error)
```

There's nothing useful  in copying these static error messages into a std::string as far as I can tell. Instead, then when calling, just check if *error == nullptr or not on the return. Yes, I know this is not in the critical path, but we don't have to do wasteful things just because it has to be STL / C++11 (C is fine in our code).

Reply:
Reasons to use std::string here:

- std::string is optimized to not dynamically allocate memory for smaller strings.
- If ops people say they need clearer error messages here that require dynamic generation, the change is localized to one function.
- It's advantageous as defensive programming to limit the use of raw pointers.

I'd like to hear the thinking of @SolidWallOfCode and @bryancall on this.

Reply:
One idea would be to pass in a `BufferWriter&` with a fixed sized buffer. The size of the error message should be easily bounded and all allocation is avoided. I don't think the short string optimization is relevant because that's only 14 characters and any meaningful error message will be longer than that.